### PR TITLE
[WIP] Add FUB HPC (Curta) config

### PR DESCRIPTION
**Pull request by wassimsalam01**
---
name: New Config
about: A new cluster config
---

Please follow these steps before submitting your PR:

- [x] If your PR is a work in progress, include `[WIP]` in its title
- [x] Your PR targets the `master` branch
- [x] You've included links to relevant issues, if any
- [x] Requested review from @nf-core/maintainers and/or #request-review on slack

Steps for adding a new config profile:

- [x] Add your custom config file to the `conf/` directory
- [x] Add your documentation file to the `docs/` directory
- [x] Add your custom profile to the `nfcore_custom.config` file in the top-level directory
- [x] Add your custom profile to the `[README.md](http://readme.md/)` file in the top-level directory
- [x] Add your profile name to the `profile:` scope in `.github/workflows/main.yml`

Marked this as a WIP as there might later be ways to improve it for more specific case usage.

### CODE REVIEW

**Comment by pontus** [comment]
Looks good, maybe specify `max_cpus` as well?

**Comment by wassimsalam01** [comment]
> Looks good, maybe specify `max_cpus` as well?

Hello Pontus. I can add this for sure. I thought it may be OK to ommit it since the number of cores a person can use is controlled by Slurm and depends on `--qos=standard`?

On a second note, could you please help me understand the linting error? The error log unfortunately doesn't give too much insight and I'm new to this.

Thank you!

**Comment by jfy133** [comment]
Hi @wassimsalam01 the linting error is for markdown formatting

(https://github.com/nf-core/configs/blob/master/.prettierrc.yml

You need to run prettier

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Comment by wassimsalam01** [comment]
> Hi @wassimsalam01 the linting error is for markdown formatting
> 
> (https://github.com/nf-core/configs/blob/master/.prettierrc.yml
> 
> You need to run prettier, or the sneaky little bot below to do it for you ;)

I see..! Thank you James 🙂

File: docs/fub_curta.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# nf-core/configs: Freie Universiät Berlin High-Performance Computer (Curta) Configuration
+
+> **NB:** In order to run pipelines using this HPC cluster, you must first apply for access [here](https://ssl2.cms.fu-berlin.de/fu-berlin/en/sites/high-performance-computing/PM_Zugang-beantragen/index.html).
+
+This profile is configured to run with Singularity version `1.1.9-1.el7`, which does not need to be loaded.
```
Comment (jfy133) [inline docs/fub_curta.md:5]: What dou mean by 'does not need to be loaded'? As in it's globally avaialble on the cluster to all users already (without any e.g. `module load` stuff)?

File: docs/fub_curta.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+Nextflow should be loaded before use. Nextflow version `22.10.6.5844` is installed. The following should be added to your sbatch/SLURM script:
+
+```
```
Comment (jfy133) [inline docs/fub_curta.md:9]:
```suggestion
```bash
```

File: conf/fub_curta.config [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    autoMounts = true
+}
+
+cleanup = true
+
+profiles {
+    debug {
+        cleanup = false
+    }
+}
```
Comment (jfy133) [inline conf/fub_curta.config:31]: This is not needed anymore, it's integrated in the main pipeline nf-core template taht most pipelines should have (some configs still have it as legacy)

**Review by jfy133 — APPROVED** [review]
A few minor things but I thjink it LGTM!

File: conf/fub_curta.config (hunk 3 above)
Comment (wassimsalam01) [reply conf/fub_curta.config]: I see. Thanks for the info. I've removed it then 👌🏼

File: docs/fub_curta.md (hunk 1 above)
Comment (wassimsalam01) [reply docs/fub_curta.md:5]: Yes that's exactly what I meant. Perhaps I should word it better to make it clearer 👍🏼

**Comment by pontus** [comment]
> Hello Pontus. I can add this for sure. I thought it may be OK to ommit it since the number of cores a person can use is controlled by Slurm and depends on `--qos=standard`?

Sorry for not getting back earlier, but the limit from the QOS is a different thing (and not visible within nextflow, where the number of cores is sometimes used). In practice, whatever the pipeline has as default will probably be used here.

**Comment by wassimsalam01** [comment]
> > Hello Pontus. I can add this for sure. I thought it may be OK to ommit it since the number of cores a person can use is controlled by Slurm and depends on `--qos=standard`?
> 
> Sorry for not getting back earlier, but the limit from the QOS is a different thing (and not visible within nextflow, where the number of cores is sometimes used). In practice, whatever the pipeline has as default will probably be used here.

No worries 🙂 thanks for your reply and explanation. I'll be sure to add it 👍🏼

**Comment by wassimsalam01** [comment]
@nf-core-bot fix linting

**Comment by jfy133** [comment]
@wassimsalam01 is this now still in WIP? Otherwise you can merge (or I can if you don't have permissions)

**Comment by wassimsalam01** [comment]
> @wassimsalam01 is this now still in WIP? Otherwise you can merge (or I can if you don't have permissions)

No I think it's ready to merge. Please go ahead 🙂 I don't have permissions

**Pull request merged**